protected/controllers/CsvController.php

```diff
@@ -0,0 +1,240 @@
+            $this->render(
+                'upload',
+                array(
+                    'errors' => $errors,
+                    'context' => 'patients',
+                )
+            );
+        }
+    }
+
+    public function createNewPatient($patient)
+    {
+        $errors = array();
+
+        if(isset($patient['CERA_number'])){
+            $new_patient = Patient::model()->findByAttributes(array('hos_num' => $patient['CERA_number']));
+            if ($new_patient !== null){
+                return $errors;
+            }
+        }
+
+        $contact = new Contact();
+        $contact_cols = array(
+            array('var_name' => 'nick_name'       , 'default' => null,),
+            array('var_name' => 'primary_phone'   , 'default' => null,),
+            array('var_name' => 'title'           , 'default' => null,),
+            array('var_name' => 'first_name'      , 'default' => null,),
+            array('var_name' => 'last_name'       , 'default' => null,),
+            array('var_name' => 'maiden_name'     , 'default' => null,),
+            array('var_name' => 'qualifications'  , 'default' => null,),
+            array('var_name' => 'contact_label_id', 'default' => null,),
+        );
+
+        foreach ($contact_cols as $col){
+            $contact->$col['var_name'] = isset($patient[$col['var_name']]) ? $patient[$col['var_name']] : $col['default'];
+        }
+
+        if(!$contact->save()){
+            return $contact->getErrors();
+        }
+
+        $address = new Address();
+        $address_cols = array(
+            array('var_name' => 'address1'       , 'default' => null,),
+            array('var_name' => 'address2'       , 'default' => null,),
+            array('var_name' => 'city'           , 'default' => null,),
+            array('var_name' => 'postcode'       , 'default' => null,),
+            array('var_name' => 'county'         , 'default' => null,),
+            array('var_name' => 'country_id'     , 'default' => 15,),
+            array('var_name' => 'email'          , 'default' => null,),
+            array('var_name' => 'date_start'     , 'default' => null,),
+            array('var_name' => 'date_end'       , 'default' => null,),
+            array('var_name' => 'address_type_id', 'default' => null,),
+        );
+
+        foreach ($address_cols as $col){
+            $address->$col['var_name'] = isset($patient[$col['var_name']]) ? $patient[$col['var_name']] : $col['default'];
+        }
+        $address->contact_id = $contact->id;
+
+        if(!$address->save()){
+            return $address->getErrors();
+        }
+
+        $new_patient = new Patient();
+        $patient_cols = array(
+            array('var_name' => 'pas_key'                       , 'default' => null,),
+            array('var_name' => 'dob'                           , 'default' => null,),
+            array('var_name' => 'gender'                        , 'default' => 'U',),
```

Comment:
Strange that the gender table isn't used at all...

Reply:
I;'d say that the table usage was their original intention, but they shelved it.